spec/services/checksum_validator_spec.rb

```diff
@@ -1,7 +1,126 @@
 
 RSpec.describe ChecksumValidator do
-  it 'has some behaviors' do
-    skip 'we should figure out what they are and test them'
+  let(:endpoint_name) { "fixture_sr3" }
+  let(:endpoint) { Endpoint.find_by(endpoint_name: endpoint_name) }
+  let(:object_dir) { "#{endpoint.storage_location}/#{DruidTools::Druid.new(druid).tree.join('/')}" }
+
+  context '#initialize' do
+    it 'sets attributes' do
+      druid = 'bj102hs9687'
+      cv = described_class.new(druid, endpoint_name)
+      expect(cv.druid).to eq "druid:#{druid}"
+      expect(cv.endpoint).to eq endpoint
+      expect(cv.checksum_results).to be_an_instance_of AuditResults
+    end
+  end
+
+  context '#validate_manifest_inventories' do
+    let(:druid) { 'bj102hs9687' }
+    let(:cv) { described_class.new(druid, endpoint_name) }
+
+    it 'instantiates moab_storage_object from druid and druid_path' do
+      expect(Moab::StorageObject).to receive(:new).with(cv.druid, a_string_matching(object_dir)).and_call_original
+      cv.validate_manifest_inventories
+    end
+
+    it 'calls validate_manifest_inventory for each moab_version' do
+      sov1 = instance_double(Moab::StorageObjectVersion)
+      sov2 = instance_double(Moab::StorageObjectVersion)
+      sov3 = instance_double(Moab::StorageObjectVersion)
+      version_list = [sov1, sov2, sov3]
+      moab_storage_object = instance_double(Moab::StorageObject, version_list: [sov1, sov2, sov3])
+      allow(cv).to receive(:moab_storage_object).and_return(moab_storage_object)
+      version_list.each do |moab_version|
+        expect(cv).to receive(:validate_manifest_inventory).with(moab_version)
+      end
+      cv.validate_manifest_inventories
+    end
+
+    it 'calls AuditResults.report_results' do
+      results = instance_double(AuditResults, add_result: nil, :actual_version= => nil, :check_name= => nil)
+      allow(AuditResults).to receive(:new).and_return(results)
+      expect(results).to receive(:report_results)
+      cv.validate_manifest_inventories
+    end
+
+    context 'file checksums in manifestInventory.xml do not match' do
+      it 'adds a MOAB_FILE_CHECKSUM_MISMATCH result' do
+        druid = 'jj925bx9565'
+        object_dir = "#{endpoint.storage_location}/#{DruidTools::Druid.new(druid).tree.join('/')}"
+        file_path1 = "#{object_dir}/v0001/manifests/versionAdditions.xml"
+        file_path2 = "#{object_dir}/v0002/manifests/versionInventory.xml"
+        results = instance_double(AuditResults, report_results: nil, :check_name= => nil)
+        allow(AuditResults).to receive(:new).and_return(results)
+        cv = described_class.new(druid, endpoint_name)
+        expect(results).to receive(:add_result).with(
+          AuditResults::MOAB_FILE_CHECKSUM_MISMATCH, file_path: a_string_matching(file_path1), version: "v1"
+        )
+        expect(results).to receive(:add_result).with(
+          AuditResults::MOAB_FILE_CHECKSUM_MISMATCH, file_path: a_string_matching(file_path2), version: "v2"
+        )
```

Comment:
i might be missing something obvious, but i wasn't clear on what caused the checksum failure in this case.

Reply:
We created a storage_root: `fixture_sr3` which contains `spec/fixtures/checksum_root01/.....` instead of `storage_root01` and storage_root02` so in all the fixtures that are in `fixture_sr3` i manually created all the errors. So if we go to the druid `jj925bx9565` and look at the `manInv` for v1 and v2 as well as versionsAddition.xml for v1 and versionInventory.xml for v2 we will see I added some `666's` to make the `CHECKSUM_MISMATCH`

Reply:
talked F2F.

Reply:
summary:  the file whose the checksum comparison failed (`versionInventory.xml`) was modified from its original version, the modification wasn't obvious to me when i was reviewing and i was too lazy to run the checksum myself, so a comment was added to the modified file for clarity.